sw/device/lib/dif/dif_i2c.h

```diff
@@ -244,7 +289,88 @@ typedef enum dif_i2c_fmt {
 } dif_i2c_fmt_t;
 
 /**
- * Computes timing parameters for an I2C device and stores them in `config`.
+ * Flags representing the status of an I2C block
+ */
+typedef struct dif_i2c_status {
+  /**
+   * Enable Host, I2C block has been initialized and enabled to act as a host,
+   * consuming entries from the FMT FIFO to perform I2C transactions, and
+   * writing the results of I2C Reads to the RX FIFO
+   */
+  bool enable_host;
+  /**
+   * Enable Target, I2C block has been initialized and enabled to act as a
+   * target device, using the TX FIFO to respond to I2C Reads and the ACQ FIFO to
+   * store data recieved from I2C Writes
+   */
+  bool enable_target;
```

Comment:
Good, some may not recognize that these can be enabled simultaneously. Some devices do two-way communication via master writes only (e.g. IPMI).

Reply:
Yes... "some" may not recognize that... and may think this portion was only included in the dif to exercise fault conditions. "Some"... may think and that when a driver was written it would handle these modes exclusively. "Some"... wouldn't have understood why there were 4 FIFOs when 2 were in use at any given time, but just may have shrugged and thought the designers of this module must have known better. "Some" people... definitely not me when I wrote this. I knew all along that both are sometimes used...

If that's a behavior we're going to rely on, I think we need to write up a functional test for it, because I bet that's a place where things get "exciting".

Reply:
Making it available to exercise fault conditions would be a good idea too!

I don't think we'd totally be able to handle the IPMB case, for what it's worth. At least the spec doesn't call out support for the "general call" reserved address.

Reply:
IDK enough about the IPMB case, but the I2C device can be configured to listen for a pair of address specifiers, so one could be set to listen for writes to address 0x00 and an appropriate set of specific addresses.